Don't package static library

ojwb opened the pull request:
It's not used, and there's a pending change to endlessm/xapian-core
which will mean we no longer build it - the packaging will stop
working when that lands (but stopping packaging it first should be
fine).

ptomato commented on debian/libxapianVERSION-dev.install:
```diff
@@ -1,7 +1,6 @@
 usr/bin/xapian-config-1.3
 usr/include
 usr/lib/cmake/xapian
-usr/lib/libxapian-1.3.a
```
I'm a bit at a loss as to what this file even does. Maybe it can be removed?

@wgreenberg You and Jasper were the last ones to touch the debian packaging, do you remember?

cosimoc replied:
The file is also part of the upstream debian packaging, so I think this is OK.

ojwb commented:
In the upstream-Debian packaging, `debian/libxapianVERSION-dev.install` is a template used to generate `debian/libxapian-dev.install`, and contains `@VERSION_SUFFIX@` placeholders which get replaced by a make rule in `debian/rules` with an empty string for stable versions and e.g. `-1.3` for development versions.

In the endless packaging, someone's committed the usually generated `debian/libxapian-dev.install`.  They've also replaced the placeholders in the template and replaced `debian/rules` with a minimal one which just delegates everything to `dh`, so the template isn't really a template and the code to use it isn't there!

[xapian-core's packaging predates `dh`, and while it'd be nice to convert it I've not had a chance yet - a minimal `dh` version won't work in Debian as the x86 baseline doesn't include SSE2 support (which xapian-core defaults to using to avoid problems with x87 excess precision, and for some extra performance) so we build two shared libraries for x86 and install them such that the dynamic linker selects the SSE2 or non-SSE2 one depending on the CPU in use; ptomato said on IRC that you don't support 32-bit x86 now, so losing this shouldn't be an issue for endless.]

I went for minimal changes in this PR rather than trying to change this back to work as upstream - the packaging as-is clearly works for you right now, so I think it makes more sense to resync the packaging with upstream when we move on from 1.3.x.

I think there's actually no need to remove the `.a` file from the template as it's never used, but it seemed better to as leaving it there just risks confusing someone looking at this later.
